src/__tests__/setup.ts

```
@@ -0,0 +1,425 @@
+  readonly exists: (path: string) => Effect.Effect<boolean>
+  readonly mkdir: (path: string) => Effect.Effect<void, { _tag: 'MakeDirectoryError', path: string, cause: unknown }>
+  readonly symlink: (target: string, linkPath: string) => Effect.Effect<void, { _tag: 'SymlinkError', pointer: string, symlink: string, cause: unknown }>
+  readonly unlink: (path: string) => Effect.Effect<void, { _tag: 'UnlinkError', symlink: string, cause: unknown }>
+  readonly readdir: (path: string) => Effect.Effect<string[], { _tag: 'ReadDirectoriesError', path: string, cause: unknown }>
+  readonly rm: (path: string, options?: { recursive?: boolean, force?: boolean }) => Effect.Effect<void, { _tag: 'RemoveEntryError', path: string, cause: unknown }>
+  readonly lstat: (path: string) => Effect.Effect<{ isSymbolicLink: () => boolean }, { _tag: 'StatError', path: string, cause: unknown }>
+  readonly rename: (oldPath: string, newPath: string) => Effect.Effect<void, { _tag: 'MoveEntryError', from: string, to: string, cause: unknown }>
+  readonly reset: () => Effect.Effect<void>
+}
+
+export class MockFileSystemService extends Context.Tag('MockFileSystemService')<
```

Biome (2.1.2)

[error] This class is unsafely merged with an interface.

The interface is declared here.

The TypeScript compiler doesn't check whether properties defined in the interface are initialized in the class.

(lint/suspicious/noUnsafeDeclarationMerging)